### PR TITLE
Special games, add ore pattern  in mixed ore

### PR DESCRIPTION
**Pull request by PlayerNoon**
### Brief description of the changes:
Add another funny ore type in special panel > mixed ore
image
image

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

File: maps/biter_battles_v2/mixed_ore_map.lua [hunk 1]
```diff
@@ -50,11 +50,45 @@ local function mixed_ore(surface, left_top_x, left_top_y)
...
+	local space = global.special_games_variables['mixed_ore_map']['size']
+	for x = 0, 31, 1 do
+		for y = 0, 31, 1 do
+			local pos = {x = left_top_x + x, y = left_top_y + y}
+			if surface.can_place_entity({name = "iron-ore", position = pos}) then
+				local ore				
+				local cx = pos.x % (space*2)
+				local cy = pos.y % (space*2)
+				if ( cx == 0 and cy == 0 ) then
+					ore = ores[2]
```
Comment (clifffrey) [inline maps/biter_battles_v2/mixed_ore_map.lua:65]: you could be evil and make there be no pattern... i.e.
`ore = ores[(pos.x * largeprime1 + pos.y * largeprime2) % 4 + 1]`
or somesuch, and you could change `space*2` to just `space` above... but probably that is too evil.

**Review by clifffrey — APPROVED** [review]

**Review by clifffrey — CHANGES_REQUESTED** [review]
I'd love if you responded to the comments, even if it is just to say "thanks, but I'm not making that change"

File: maps/biter_battles_v2/mixed_ore_map.lua [hunk 2]
```diff
@@ -50,11 +50,45 @@ local function mixed_ore(surface, left_top_x, left_top_y)
...
+				elseif ( cx == 0 and cy == space ) then
+					ore = ores[3]
+				elseif	( cx == space  and cy == 0 ) then
+					ore = ores[4]
+				elseif	( cx == space  and cy == space  ) then											
+					ore = ores[5]
+					if math.random(1,1000) > 998 then
```
[the remark this answers is not in the document]
Comment (PlayerNoon) [reply maps/biter_battles_v2/mixed_ore_map.lua:72]: nice idea. but i try to follow how the code is already written, everyone uses integers so why change it.

File: maps/biter_battles_v2/mixed_ore_map.lua (hunk 1 above)
Comment (PlayerNoon) [reply maps/biter_battles_v2/mixed_ore_map.lua:65]: i don't understand your code, i don't know how to apply it so can't test it.

**Comment by PlayerNoon** [comment]
why my responses for your comments are important? i don't have experience so don't understand. can't merge it without? do i have to resolve all conversations? or just to have a nice conversation? :D

**Comment by clifffrey** [comment]
> why my responses for your comments are important? i don't have experience so don't understand. can't merge it without? do i have to resolve all conversations? or just to have a nice conversation? :D

Great question!

If you don't respond, then I don't know if you have read the comment or not, and I don't know if we communicated successfully or not.

In the end, it really is important that the code does more than "just work" - because this is not the last change that we will make. Other people will also make changes to the code in the future, so anything we can do to make it more understandable/better/more-consistent is worthwhile, even if it means that making one change takes a bit longer.

**Review by clifffrey — APPROVED** [review]

**Pull request merged**